usr/src/uts/armv8/io/gfx_private/gfxp_devmap.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
+ * Use is subject to license terms.
+ */
+
+#include <sys/debug.h>
+#include <sys/types.h>
+#include <sys/param.h>
+#include <sys/time.h>
+#include <sys/buf.h>
+#include <sys/errno.h>
+#include <sys/systm.h>
+#include <sys/conf.h>
+#include <sys/signal.h>
+#include <vm/page.h>
+#include <vm/as.h>
+#include <vm/hat.h>
+#include <vm/seg.h>
+#include <vm/seg_dev.h>
+#include <vm/hat_aarch64.h>
+#include <sys/ddi.h>
+#include <sys/devops.h>
+#include <sys/sunddi.h>
+#include <sys/ddi_impldefs.h>
+#include <sys/fs/snode.h>
+#include <sys/pci.h>
+#include <sys/vmsystm.h>
+#include <sys/int_fmtio.h>
+#include <sys/gfx_private.h>
+
+#ifdef __xpv
```

Comment:
I've been on the fence about this, and I still am (which means all the existing code is inconsistent).

For `__xpv` at least, though, I think we should just remove all this.  Xen or aarch64 support is unlikely, and the ancient code we have being helpful probably even less likely.

You should just be able to `unidef -U__xpv`, if you agree with me